ext/drbbdup/drbbdup.h

```diff
@@ -0,0 +1,334 @@
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef _DRBBDUP_H_
+#define _DRBBDUP_H_ 1
+
+#include "drmgr.h"
+#include <stdint.h>
+
+/**
+ * @file drbbdup.h
+ * @brief Header for DynamoRIO Basic Block Extension
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \addtogroup drbbdup Basic Block Duplicator
+ */
+/*@{*/ /* begin doxygen group */
+
+/** Success code for each drbbdup operation */
+typedef enum {
+    DRBBDUP_SUCCESS,                       /**< Operation succeeded. */
+    DRBBDUP_ERROR_INVALID_PARAMETER,       /**< Operation failed: invalid parameter. */
+    DRBBDUP_ERROR_CASE_ALREADY_REGISTERED, /**< Operation failed: already registered. */
+    DRBBDUP_ERROR_CASE_LIMIT_REACHED,      /**< Operation failed: case limit reached. */
+    DRBBDUP_ERROR_ALREADY_INITIALISED,     /**< DRBBDUP can only be initialised once. */
+    DRBBDUP_ERROR,                         /**< Operation failed. */
+} drbbdup_status_t;
+
+/***************************************************************************
+ * User call-back functions
+ */
+
+/**
+ * Set up initial information related to a managing copies of a new basic block.
+ * A pointer-sized value indicating the default case encoding is returned.
+ * The boolean value written to \p enable_dups specifies whether code duplication should
+ * be done for this basic block. If false, the basic block is always executed under the
+ * default case and no duplications are made.
+ *
+ * Use drbbdup_register_case_value() to register other case encodings.
+ *
+ * @param[in] enable_dups indicates whether basic block code is duplicated to support
```

Comment:
I see that drreg avoids using @params for insert API functions (inc. drcontext, ilist, where). I think I should do the same rather than documenting again what is pretty standard in DR.

Reply:
Done